src/scope.rs

```diff
@@ -1618,6 +1888,75 @@ mod raw {
     }
   }
 
+  #[repr(C)]
+  #[derive(Debug)]
+  pub(super) struct DisallowJavascriptExecutionScope([usize; 3]);
+
+  impl DisallowJavascriptExecutionScope {
+    /// This function is marked unsafe because the caller must ensure that the
+    /// returned value isn't dropped before `init()` has been called.
+    pub unsafe fn uninit() -> Self {
+      // This is safe because there is no combination of bits that would produce
+      // an invalid `[usize; 3]`.
```

Comment:
Although this follows the example of other scopes, this isn't actually safe now that I look at it: uninitialized memory is different conceptually than any (initialized) combination of bits, and uninitialized memory isn't a valid value for `[usize; 3]`. It should be this instead:

```rust
pub(super) struct DisallowJavascriptExecutionScope(MaybeUninit<[usize; 3]>);
```

This doesn't block this PR, though.

Reply:
The Rust docs [now mention](https://doc.rust-lang.org/std/mem/union.MaybeUninit.html#initializing-an-array-element-by-element) that creating an array of `MaybeUninit<T>` with `MaybeUninit::uninit().assert_init()` is safe, so this is not actually an issue.